web/components/new-contract/new-contract-panel.tsx

```diff
@@ -50,6 +51,10 @@ export function NewContractPanel(props: {
     params?.outcomeType ? 'filling contract params' : 'choosing contract'
   )
 
+  // Don't use the usePersistentLocalState hook for this, because there's too high a risk that it will survive in local storage
+  // longer than it should under a trivial paramsKey like '', and improperly prevent users from creating any new contracts.
+  const [idempotencyKey] = useState(randomString())
```

Comment:
I think it'd be fine to just add this key to the `ContractParamsForm` component rather than pass the prop down.

Reply:
Sure, I'll do that. 

My purpose for defining it here is to make the idempotencyKey stable even if the user goes back and re-chooses the contract type, which they may likely do when banging around to try and make an error go away. If you feel that's a compelling reason, I'll put it back.

Reply:
Oh I see, as long as we provide them with the url of their newly minted contract I bet they won't go banging around.

Reply:
Yeah fair point! I did add it to the ContractParamsForm component rather than pass the prop down, per your suggestion. Does it look good at this point?